### PR TITLE
weechat: fix Ruby option

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

🙃.

### CODE REVIEW

File: Formula/weechat.rb [hunk 1]
```diff
@@ -26,7 +35,7 @@ class Weechat < Formula
   depends_on "lua" => :optional
   depends_on "perl" => :optional
   depends_on "python" => :optional
-  depends_on "ruby" => :optional if MacOS.version <= :sierra
```
Comment (ilovezfs) [inline Formula/weechat.rb]: why?

Comment (DomT4) [reply Formula/weechat.rb]: You didn't really explain in the other PR _why_ you'd capped this below Sierra, so, you tell me? 🙈

Comment (DomT4) [reply Formula/weechat.rb]: The problem remains either way. Homebrew's Ruby is at 2.5.0, which `weechat` stable has no idea exists before this PR. But I can find no fault building against Ruby on High Sierra other than that.

Comment (ilovezfs) [reply Formula/weechat.rb]: The minimum version was set to 2.1

Comment (DomT4) [reply Formula/weechat.rb]: I'll look around but this proved an exceptionally stubborn issue in the past.

Comment (ilovezfs) [reply Formula/weechat.rb]:
```
RUBY_EXECUTABLE:FILEPATH=/usr/bin/ruby
RUBY_INCLUDE_DIRS:PATH=/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/include/ruby-2.3.0
RUBY_LIB:FILEPATH=/usr/lib/libruby.dylib
```

Comment (ilovezfs) [reply Formula/weechat.rb]: looks like `-DRUBY_EXECUTABLE=/usr/bin/ruby` is sufficient

Comment (DomT4) [reply Formula/weechat.rb]: Alas, it isn't.

```
==> Summary
🍺  /usr/local/Cellar/weechat/2.0.1: 40 files, 5.9MB, built in 24 seconds
~> otool -L /usr/local/Cellar/weechat/2.0.1/lib/weechat/plugins/ruby.so
/usr/local/Cellar/weechat/2.0.1/lib/weechat/plugins/ruby.so:
	/usr/local/opt/ruby/lib/libruby.2.5.dylib (compatibility version 2.5.0, current version 2.5.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
```

Comment (DomT4) [reply Formula/weechat.rb]:
```diff
diff --git a/Formula/weechat.rb b/Formula/weechat.rb
index fabaa5dd1a..40ff809fdd 100644
--- a/Formula/weechat.rb
+++ b/Formula/weechat.rb
@@ -35,7 +35,7 @@ class Weechat < Formula
   depends_on "lua" => :optional
   depends_on "perl" => :optional
   depends_on "python" => :optional
-  depends_on "ruby" => :optional
+  depends_on "ruby" => :optional if MacOS.version <= :sierra
   depends_on "curl" => :optional
 
   def install
@@ -49,9 +49,14 @@ class Weechat < Formula
       args << "-DCMAKE_BUILD_TYPE=Debug"
     end
 
+    if build.without? "ruby"
+      args << "-DENABLE_RUBY=OFF"
+    elsif build.with?("ruby") && MacOS.version >= :sierra
+      args << "-DRUBY_EXECUTABLE=/usr/bin/ruby"
+    end
+
     args << "-DENABLE_LUA=OFF" if build.without? "lua"
     args << "-DENABLE_PERL=OFF" if build.without? "perl"
-    args << "-DENABLE_RUBY=OFF" if build.without? "ruby"
     args << "-DENABLE_ASPELL=OFF" if build.without? "aspell"
     args << "-DENABLE_TCL=OFF" if build.without? "tcl"
     args << "-DENABLE_PYTHON=OFF" if build.without? "python"
```

Comment (ilovezfs) [reply Formula/weechat.rb]: Throwing `-DRUBY_LIB=/usr/lib/libruby.dylib` at it too fixes that.

**Comment by ilovezfs** [comment]
OK, so given this, any reason to change it to `depends_on "ruby" => :optional`? Happy to do so if you still think that's preferable.

**Comment by DomT4** [comment]
I'm not aware of any major upside _(or downside, to be fair)_ to providing the unconditional option. Unless someone asks for it I'm not that fussed, now we've wangled CMake into submission.

**Comment by ilovezfs** [comment]
:rocket:

**Comment by ilovezfs** [comment]
Autotools:Latin::CMake:Esperanto

**Pull request merged**